h/tasks/indexer.py

```diff
@@ -42,8 +42,34 @@ def delete_annotation(id_):
 
 @celery.task(acks_late=False)
 def sync_annotations(limit):
-    search_index = celery.request.find_service(name="search_index")
-    search_index.sync(limit)
+    queue = celery.request.find_service(name="search_index")._queue
+
+    counts = queue.sync(limit)
+
+    log.info(dict(counts))
+    newrelic.agent.record_custom_metrics(
+        [
+            (
+                "Custom/SyncAnnotations/AnnotationsIndexed",
+                sum(
+                    [
+                        counts[queue.Result.MISSING],
+                        counts[queue.Result.DIFFERENT],
+                        counts[queue.Result.FORCED],
+                    ]
+                ),
+            ),
+            (
+                "Custom/SyncAnnotations/JobsCompleted",
+                sum(
+                    [
+                        counts[queue.Result.DELETED_FROM_DB],
+                        counts[queue.Result.UP_TO_DATE],
+                    ]
+                ),
+            ),
+        ]
+    )
```

Comment:
I don't think we want separate metrics for each different reason _why_ an annotation was indexed or a job completed, in New Relic. Those separate metrics are useful in the logs where I use them to observe individual runs of the task and exactly what they did when testing or debugging. New Relic is a more "zoomed out" view and I think it's more useful just to have sum metrics of how many annotations were indexed or how many jobs were completed.

Alternatively we could do this summing in `Queue.sync()` instead and return the summed numbers in the counter along with the source numbers.

Or we could report the separate metrics to New Relic and do the summing in New Relic. So the "indexing rate" graph in New Relic would be a graph of the sum of three metrics.

Reply:
Note that you can't distinguish annotations that were indexed by the sync task because the API failed to index them, from annotations that were indexed by an admin page such as the rename-user page. To distinguish those I think we'd need `Queue.sync()` to return separate tag-based counts, and I don't think we should bother.